lib/katello/tasks/unify_hosts.rake

```diff
@@ -0,0 +1,95 @@
+namespace :katello do
+  desc "Unify hosts that have registered with both fqdn and shortname.\
+        Run with HOSTS=foobar.example.com,foobar to run for specific hosts."
+  task :unify_hosts => :environment do
+    def main
+      hostnames = ENV['HOSTS'].try(:split, ',')
+      if hostnames.blank?
+        unify_all
+      elsif hostnames.count == 2
+        unify_two(hostnames[0], hostnames[1])
+      else
+        puts "HOSTS specified but did not receive two. Usage: HOSTS=foobar.example.com,foobar"
+      end
+    end
+
+    def unify_all
+      ::Host.unscoped.where("name like '%.%'").find_each do |host|
+        short_host = find_match(host)
+        unify(host, short_host) if short_host
+      end
+    end
+
+    def unify_two(name1, name2)
+      host1 = Host.find_by(:name => name1)
+      host2 = Host.find_by(:name => name2)
+      if host1.nil? || host2.nil?
+        puts "Could not find both hosts #{host1} and #{host2}\n"
+      elsif name1.starts_with?(name2)
```

Comment:
In this case the user would have to have passed in  HOSTS=foobar.example.com,foobar.example.com in which case it'd be the same host.  I can check for that input and throw an error